### DISCUSSION TITLE
Angularアプリのディレクトリ構成どうしてますか？

### DISCUSSION THREAD

**Discussion by komura-c** [StartupAngular/discussions#17] [General]

**Comment by rettar5** [comment]
セッション1

**Reply by rettar5** [reply]
## ディレクトリ構成どうしてますか
- 3アプリをモノレポ管理
- 1プロダクトで、立ち上げたやつがそのまま使われてる
  - 整えていきたい
  - sharedフォルダを作ろうとしてる
- 3アプリあって、別々のリポジトリで管理
  - 最初に立ち上げたもの使ってる
　　  - pages, componentsなどで分けてる
  - sharedは分けるのが難しい
    - 整理分類のところ
    - moduleかができてない状態
    - 分けるのが課題になっている
- 1つのプロダクトから切り出せない固有の機能などがあり、ドメインしならないと切り離せないのが課題

## 俺が考えた最強のshared
- sharedの下にpresentationを作り、ボタンやラベルを入れていく
  - テストカバレッジ100%にする
  - それ以外のモデルやサービス系も100%にしたい
  - 現状テストが書かれてないので、担保されているコンポーネントを増やしていきたい
- sharedみたいなディレクトリは作らないほうがいいかも
  - Single Component Angular Moduleという考え方がある
